actions/inventory.go

```diff
@@ -1,54 +1,136 @@
+// The inventory package ties together the various collectors under utils through its InventoryCollectorAction type.
 package actions
 
 import (
 	"context"
-	"log"
+	"runtime/debug"
 	"strings"
 
 	"github.com/bmc-toolbox/common"
+	"github.com/pkg/errors"
 	"github.com/r3labs/diff/v2"
 	"golang.org/x/exp/slices"
 
+	"github.com/metal-toolbox/ironlib/model"
 	"github.com/metal-toolbox/ironlib/utils"
-	"github.com/pkg/errors"
 )
 
-// The inventory package ties together the various collectors under utils
-// using the interfaces defined utils.interface.
-
 var (
 	ErrInventoryDeviceObjNil = errors.New("method Inventory() expects a valid device object, got nil")
+	ErrPanic                 = errors.New("recovered from panic")
 )
 
+// InventoryCollectorAction provides methods to collect hardware, firmware inventory.
+type InventoryCollectorAction struct {
+	// collectors registered for inventory collection.
+	collectors Collectors
+
+	// device is the model in which the collected inventory is recorded.
+	device *common.Device
+
+	// Enable trace logging on the collectors.
+	trace bool
+
+	// a.failOnError when enabled causes ironlib to return an error
+	// if a collector returns an error
+	failOnError bool
+
+	// dynamicCollection when enabled will cause ironlib to
+	// enable collectors based on the detected component vendor.
+	dynamicCollection bool
+
+	// disableCollectorUtilities is the list of collector utilities
+	// to be disabled, this is the name of collector utility
+	// which is returned by its Attributes() method.
+	disableCollectorUtilities []model.CollectorUtility
```

Comment:
Maybe this should be 'disabledCollectorUtilities' ?

Reply:
sounds right, fixed in[2057e09](https://github.com/metal-toolbox/ironlib/pull/75/commits/2057e0930e7282cb218d0230f39567138b082fd3)